zipline/finance/performance/period.py

```diff
@@ -203,9 +200,12 @@ def calculate_performance(self):
 
     def record_order(self, order):
         if self.keep_orders:
-            dt_orders = self.orders_by_modified[order.dt]
-            if order.id in dt_orders:
-                del dt_orders[order.id]
+            try:
+                dt_orders = self.orders_by_modified[order.dt]
+                if order.id in dt_orders:
+                    del dt_orders[order.id]
```

Comment:
sometimes the ordering will mess up the test harness. I had something in blotter which didn't need to be ordered, but due to the SlippageModel semantics ended up having different algo exhaust.